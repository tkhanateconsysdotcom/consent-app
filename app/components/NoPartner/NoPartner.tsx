import React from "react";
import { ScrollView, View } from "react-native";
import { useNavigation } from "@react-navigation/native";

import { t } from "utils";
import { navigate } from "navigation";

import Text from "../Text";
import StickersImage from "../StickersImage";
import Button from "../Button";
import styles from "./NoPartner.styles";

export default function NoPartner() {
  const navigation = useNavigation();
  const navigator = navigate(navigation);

  return (
    <ScrollView style={styles.container}>
      <StickersImage sticker="earth" />
      <View style={styles.textView}>
        <Text.H1 style={styles.header}>
          {t("NO_PARTNER_COMPONENT_TITLE")}
        </Text.H1>
        <Text.Primary style={styles.paragraph}>
          {t("NO_PARTNER_COMPONENT_THANKS")}
        </Text.Primary>
        <Text.Primary style={styles.paragraph}>
          {t("NO_PARTNER_COMPONENT_START_USING_APP")}
        </Text.Primary>
        <Button.Primary
          fullWidth
          style={styles.button}
          onPress={() => navigator.openAddPartner()}
          textType={"Primary"}
        >
          <Text.Primary bold white>
            {t("NO_PARTNER_COMPONENT_ADD_FIRST_PARTNER")}
          </Text.Primary>
        </Button.Primary>
      </View>
    </ScrollView>
  );
}
