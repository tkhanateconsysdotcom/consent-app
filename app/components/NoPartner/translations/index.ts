import en from "./en.json";
import fr from "./fr.json";
import de from "./de.json";
import sv from "./sv.json";
import es from "./es.json";
import pt from "./pt.json";
import da from "./da.json";
import ru from "./ru.json";
import pl from "./pl.json";

interface TranslationKeys {
  NO_PARTNER_COMPONENT_TITLE: string;
  NO_PARTNER_COMPONENT_THANKS: string;
  NO_PARTNER_COMPONENT_START_USING_APP: string;
  NO_PARTNER_COMPONENT_ADD_FIRST_PARTNER: string;
}

export { en, fr, de, sv, es, pt, da, ru, pl, TranslationKeys };
