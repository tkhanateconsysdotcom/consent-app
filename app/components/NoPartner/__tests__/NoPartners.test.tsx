import React from "react";
import { create } from "react-test-renderer";

import NoPartner from "../";

jest.unmock("../");

it("renders correctly NoPartner", () => {
  const tree = create(<NoPartner />).toJSON();
  expect(tree).toMatchSnapshot();
});
