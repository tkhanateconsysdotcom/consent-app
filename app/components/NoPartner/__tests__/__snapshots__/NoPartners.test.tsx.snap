// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly NoPartner 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "paddingHorizontal": 14,
    }
  }
>
  <View>
    <StickersImage
      sticker="earth"
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Text.H1
        style={
          Object {
            "paddingVertical": 14,
          }
        }
      >
        Hi there ðŸ‘‹
      </Text.H1>
      <Text.Primary
        style={
          Object {
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        Thanks for joining our fight against predatory behavior!
      </Text.Primary>
      <Text.Primary
        style={
          Object {
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        To start tracking your partners, tap the button below
      </Text.Primary>
      <Button.Primary
        fullWidth={true}
        onPress={[Function]}
        style={
          Object {
            "marginVertical": 20,
          }
        }
        textType="Primary"
      >
        <Text.Primary
          bold={true}
          white={true}
        >
          Add my first partner
        </Text.Primary>
      </Button.Primary>
    </View>
  </View>
</RCTScrollView>
`;
